### PR TITLE
[Fix](Nereids) fix sql-cache for nereids.

### PR DESCRIPTION
**Pull request by dutyu**
## Proposed changes

Issue Number: close #xxx

1. should not use `((LogicalPlanAdapter)parsedStmt).getStatementContext().getOriginStatement().originStmt.toLowerCase()` as the cache key (do not invoke `toLowerCase()`), for example: `select * from tbl1 where k1 = 'a'` is different with `select * from tbl1 where k1 = 'A'`, so the cache should be missed.
2. according to [issue 6735](https://github.com/apache/doris/issues/6735) , the cache key should contains all views' s ddl sql (including nested views)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dutyu** [comment]
run buildall

**Comment by dutyu** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.87 seconds
 stream load tsv:          515 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162194631 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.22 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162378458 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/nereids/CascadesContext.java [hunk 1]
```diff
@@ -97,6 +99,7 @@ public class CascadesContext implements ScheduleContext {
     private final RuntimeFilterContext runtimeFilterContext;
     private Optional<Scope> outerScope = Optional.empty();
     private List<TableIf> tables = null;
+    private Set<View> views = Sets.newHashSet();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/CascadesContext.java:102]: move to statement context

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/LogicalPlanAdapter.java [hunk 2]
```diff
@@ -79,6 +81,10 @@ public ArrayList<String> getColLabels() {
         return colLabels;
     }
 
+    public List<View> getViews() {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/LogicalPlanAdapter.java:84]: move views to StatementContext, u could get them from StatementContext directly

**Comment by dutyu** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.41 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162316600 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/LogicalPlanAdapter.java (hunk 2 above)
Comment (dutyu) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/LogicalPlanAdapter.java:84]: fixed

File: fe/fe-core/src/main/java/org/apache/doris/nereids/CascadesContext.java (hunk 1 above)
Comment (dutyu) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/CascadesContext.java]: fixed

**Review by morrySnow — APPROVED** [review]

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**